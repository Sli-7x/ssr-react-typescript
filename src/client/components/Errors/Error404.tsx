import * as React from 'react';

class Error404 extends React.Component {
  componentWillMount() {
    const { staticContext }: any = this.props;
    if (staticContext) {
      staticContext.status = 404;
    }
  }

  render() {
    return <h1>Sorry, canâ€™t find that. 404</h1>;
  }
}

export default Error404;
