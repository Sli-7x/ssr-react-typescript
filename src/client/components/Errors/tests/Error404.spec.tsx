import * as React from 'react';
import Error404 from '../Error404';
import * as renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom';

describe('Error404.tsx', () => {
  it('should render correctly', () => {
    const filledTree = renderer
      .create(
        <StaticRouter location="/asd" context={{}}>
          <Error404 />
        </StaticRouter>,
      )
      .toJSON();
    expect(filledTree).toMatchSnapshot();
  });
});
