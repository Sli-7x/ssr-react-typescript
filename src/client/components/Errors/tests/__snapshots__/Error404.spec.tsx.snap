// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error404.tsx should render correctly 1`] = `
<h1>
  Sorry, canâ€™t find that. 404
</h1>
`;
